pkg/types/types.go

```diff
@@ -58,9 +58,13 @@ func (n *IPNet) UnmarshalJSON(data []byte) error {
 type NetConf struct {
 	Name string `json:"name,omitempty"`
 	Type string `json:"type,omitempty"`
-	IPAM struct {
-		Type string `json:"type,omitempty"`
-	} `json:"ipam,omitempty"`
+	IPAM IPAM   `json:"ipam,omitempty"`
+}
+
+type IPAM struct {
+	Type   string              `json:"type,omitempty"`
+	Subnet string              `json:"subnet"`
```

Comment:
@jonboulle - does this just mean I need to add a couple more omitempty tags plus a another field?